underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AInputStream.java

```diff
@@ -123,30 +129,59 @@ public long skip(long n) throws IOException {
     if (n <= 0) {
       return 0;
     }
-    closeStream();
     mPos += n;
-    openStream();
     return n;
   }
 
   /**
-   * Opens a new stream at mPos if the wrapped stream mIn is null.
+   * Seek if we could reuse the existing stream or create a new one.
+   * @param length the content length to be read
    */
-  private void openStream() throws IOException {
-    if (mIn != null) { // stream is already open
+  private void seekStream(int length) throws IOException {
+    if (mIn != null && mPos == mInPos) { // stream is already open and at the expected position
       return;
     }
+    if (mIn != null && mPos > mInPos) { // stream is already open but the caller is seeking forward
+      if (mPos - mInPos < mIn.available()) {
```

Comment:
we already recorded `mInPosEnd` so no need of checking mIn.available() which may not be correct?

I expect if read is out of range, then getObject will already fail?
as long as mInPos< mPos < mInPosEnd, it should exist?